gpii/node_modules/flowManager/src/Capture.js

```diff
@@ -0,0 +1,229 @@
+            args: ["{flowManager}.deviceReporter"]
+        },
+        "onSolutionsForCurrentDevice.getSolutionsPromise": {
+            funcName: "gpii.flowManager.getSolutionsPromise",
+            args: [ "{flowManager}.solutionsRegistryDataSource", "{arguments}.0"],
+            priority: "after:getDeviceContextPromise"
+        },
+        "onSolutionsForCurrentDevice.solutionsRegistryEntriesToPromise": {
+            funcName: "fluid.toPromise",
+            args: ["{arguments}.0.solutionsRegistryEntries"],
+            priority: "after:getSolutionsPromise"
+        },
+        // Begin declaration of Promise Chain for onCaptureSettingsForCurrentDevice
+        "onCaptureSettingsForCurrentDevice.getInstalledSolutionsForCurrentDevice": {
+            func: "{that}.getInstalledSolutionsForCurrentDevice"
+        },
+        "onCaptureSettingsForCurrentDevice.captureSystemSettings": {
+            func: "{that}.captureSystemSettings",
+            args: ["{arguments}.0", "{arguments}.1"],
+            priority: "after:getInstalledSolutionsForCurrentDevice"
+        },
+        "onCaptureSettingsForCurrentDevice.formatRawCapturedSettings": {
+            func: "gpii.flowManager.formatRawCapturedSettings",
+            args: ["{arguments}.0"],
+            priority: "after:captureSystemSettings"
+        }
+    },
+    invokers: {
+        getInstalledSolutionsForCurrentDevice: {
+            funcName: "gpii.flowManager.getInstalledSolutionsForCurrentDevice",
+            args: ["{that}"]
+        },
+        getSystemSettingsCapture: {
+            funcName: "gpii.flowManager.getSystemSettingsCapture",
+            args: ["{that}.events.onCaptureSettingsForCurrentDevice", "{arguments}.0"] // options
+        },
+        captureSystemSettings: {
+            funcName: "gpii.flowManager.captureSystemSettings",
+            args: ["{lifecycleManager}.read", "{arguments}.0", "{arguments}.1"] // solutionsRegistryEntries, options
+        }
+    }
+});
+
+gpii.flowManager.getInstalledSolutionsForCurrentDevice = function (that) {
+    return fluid.promise.fireTransformEvent(that.events.onSolutionsForCurrentDevice);
+};
+
+/**
+ * This main API entry point for capturing settings from a system or computer. This captures
+ * the actual settings on the device, so it assumed to be running in a local untrusted flow
+ * manager.
+ *
+ * @param {Event} onCaptureSettingsForCurrentDevice - The transforming promise chain
+ * @param {Object} options - Options for this chain.
+ * @param {Array} options.solutionsList - An array of solution IDs to filter by when
+ * retreiving settings. If this option is not included, all available settings will be
+ * returned. ex: `["com.microsoft.windows.mouseSettings", "com.freedomscientific.jaws"]`.
+ * @return {Promise} A promise resolved with the payload of captured system settings.
+ */
+gpii.flowManager.getSystemSettingsCapture = function (onCaptureSettingsForCurrentDevice, options) {
```

Comment:
Please use the namespace "gpii.flowManager.capture" for all functions rather than "gpii.flowManager".

Reply:
updated the namespaces